### ISSUE TITLE
[PROPOSAL/QUESTION] URL Protocol handler

### ISSUE THREAD

**Issue by FallenChromium** [alfredh/baresip#610]
Hello there! I found this project week ago and what I want to say - this is really awesome, there's for sure lack of lightweight and customizable open-source SIP clients on Linux. Basically, what I want to ask is the ways URL protocols handling can be implemented, as I have really small experience with C. I already seen httpd module and it works quite well, but exposing a listening port and keeping this web server alive is a bit not what I'm talking about. The `sip:`, `tel:` and `callto:` links are really convenient - they are universal so integration can be done everywhere. The problem is that proper protocol handler needs to exec something - and this would call new baresip instance. I like the way UI modules work out there - if I'll place a call via httpd, gtk_mod will show up with call control. What comes in my mind is kind of a shell wrapper, whether placing calls via standard API (as httpd does) on execution with call number as an argument, or some evdev manipulation thing (didn't got this yet, documentation is thin there :rofl: ). The question is can I implement separate command as a module to baresip? Seems controversial to me, but other way I need to "re-invent bicycle" by setting up intercepting things and so on, this seems wacky. Am I missing something some opportunity?

**Comment by alfredh** [comment]
it is possible to start baresip with a target uri to dial:

```
$ baresip -e d1234
```

or 

```
$ baresip -e "/dial sip:1234@domain.com"
```

any long/short command can be executed and you can also repeat them
many times:

```
$ baresip -e /sysinfo -e /quit
```

you can write a shell script that registers your URI handlers to the relevant
platform API, and then invoke the baresip process in your hook.

**Comment by FallenChromium** [comment]
Oh, this is actually cool! Didn't seen "-e" arg. Any way I can execute commands inside running baresip and not launching new instance? Feels more like a evdev stuff, I may end up writing shell script that is running as URl handler and passing data to event, but i didn't digged into this yet. I can't see any downsides of relaunching baresip for every call i need to place but I don't find this thing good either. Also I'm thinking about running baresip as a daemon, but then it can't launch GTK+ process (some desktop acquiring error which is expected then running as daemon), can i workaround that?

**Comment by alfredh** [comment]
if the baresip instance is running, you can use a management interface to talk
to it, such as `ctrl_tcp.so` or `cons.so`.

Example if cons.so is loaded and listening on port 5555:

```
$ echo -n d1234 | nc -u 127.0.0.1 5555
```

any command can also be sent via these management interfaces.

**Comment by FallenChromium** [comment]
Right, I'll try 'em all and report results back in this issue if someone will be interested. Thank you for your advices @alfredh!

**Comment by alfredh** [comment]
@FallenChromium if this answered your questions, please close the issue.

we are trying to keep the Issue tracker mainly for critical defects.
